src/repository/index.ts

```diff
@@ -135,15 +144,11 @@ export default class Repository extends EventEmitter implements EventEmitter {
     this.storageProvider = storageProvider;
     this.segments = new Map();
     this.eventSource = eventSource;
+    this.mode = mode;
     if (this.eventSource) {
       // On re-connect it guarantees catching up with the latest state.
       this.eventSource.addEventListener('unleash-connected', (event: { data: string }) => {
-        // reconnect
-        if (this.initialEventSourceConnected) {
-          this.handleFlagsFromStream(event);
-        } else {
-          this.initialEventSourceConnected = true;
-        }
+        this.handleFlagsFromStream(event);
```

Comment:
in streaming mode we always rely 100% on streaming. previously first request in streaming was a polling request